src/Upload.test.js

```diff
@@ -127,6 +127,24 @@ describe('Upload', () => {
           xhr,
         })
       })
+
+      describe('if headers are provided', () => {
+        const mockXHR = {
+          setRequestHeader: jest.fn(),
+          readyState: 1,
```

Comment:
I'm actually not sure if the xhr object is in readyState 1 (opened) when `directUploadWillCreateBlobWithXHR` is called. Ideas?

Reply:
The `activestorage` package makes heavy use of `setRequestHeader` before we get the `xhr`, so it should always be ready to go. https://github.com/rails/rails/blob/5-2-1/activestorage/app/javascript/activestorage/blob_record.js#L15

Checking with a breakpoint in my app, it seems to work.